pure.zsh

```diff
@@ -103,19 +103,30 @@ prompt_pure_preexec() {
 	export VIRTUAL_ENV_DISABLE_PROMPT=${VIRTUAL_ENV_DISABLE_PROMPT:-12}
 }
 
+# Change the colors if their value are different from the current ones
+prompt_pure_set_colors() {
+	local color_temp
+	for key val in ${(kv)prompt_pure_colors}; do
+		if ! zstyle -T ":prompt:pure:$key" color "$val"; then
+			zstyle -s ":prompt:pure:$key" color color_temp
+			test -n "$color_temp" && prompt_pure_colors[$key]=$color_temp
```

Comment:
I think it would be good to have a way to restore the original color if the zstyle definition is deleted (e.g. `zstyle -d prompt:pure:prompt:\*`).